### PR TITLE
Jacob create post

### PR DESCRIPTION
**Pull request by Jet004**
- Created custom model for posts with custom methods
- Added new route with login protections for creating a new post
- Added controller to handle post creation
- Added required tests

### CODE REVIEW

File: __test__/integrationTests.test.ts [hunk 1]
```diff
@@ -285,3 +285,58 @@ describe("Integration tests for AUTH routes:", () => {
...
+                url: "/api/posts/create",
+                method: "POST",
+                headers: {
+                    Cookie: cookie,
+                },
+                data: {
+                    title: "test",
+                    content: "test",
+                    status: "DRAFT",
+                    category: "GENERAL",
```
Comment (2016lisali) [inline __test__/integrationTests.test.ts:482]: should the tags be included in the data?

**Review by 2016lisali — APPROVED** [review]
Great work. Thanks, Jacob 👍

**Review by B08waffles — APPROVED** [review]
Looks good, keen to merge!

**Review by QiujuYang — APPROVED** [review]

File: __test__/integrationTests.test.ts (hunk 1 above)
Comment (Jet004) [reply __test__/integrationTests.test.ts:482]: Yes, you are right. I'll add that on Monday!

Comment (Jet004) [reply __test__/integrationTests.test.ts:482]: Tags are now inserted when a post is created. Please be aware that the way it is currently set up requires us to create a list of possible tags in the database. Only tags already in the database can be selected. We should consider some way to extend this list as people actually use the site at a future date.

**Review by B08waffles — APPROVED** [review]

**Review by 2016lisali — APPROVED** [review]
Looks great, well done, Jacob :)

**Pull request merged**